docs-src/0.6/src/reference/component_props.md

```diff
@@ -151,3 +151,185 @@ DemoFrame {
+```
+
+```inject-dioxus
+DemoFrame {
+    component_extending_elements::App {}
+}
+```
+
+No need to manually forward every attribute. Simply spread `..attrs`, and your component becomes fully HTML-compatible.
+
+### Merging Multiple Sources
+
+You can extend multiple attribute domains (e.g. global + tag-specific):
+
+```rust, no_run
+{{#include src/doc_examples/component_extending_elements.rs:ExtendingButton}}
+```
+
+```rust, no_run
+{{#include src/doc_examples/component_extending_elements.rs:ExtendingButtonUsage}}
+```
+
+```inject-dioxus
+DemoFrame {
+    component_extending_elements::ActionApp {}
+}
+```
+
+Dioxus resolves conflicts internally. However, avoid combining elements with **incompatible attribute semantics**.
+
+### Caveats and Tips
+
+- **Attribute collision resolution**: If multiple values are passed for the same attribute (e.g. `class` appears in both props and `..attrs`), the **last occurrence wins**. This is standard HTML behavior, but it's worth watching out for when merging attributes from different sources.
+
+- **Cannot override named props**: The `extends` system only covers HTML attributes. You **cannot override custom or named props** declared explicitly in the component signature. These must still be handled and destructured manually.
+
+  ```rust, no_run
+  #[component]
+  fn Card(
+      title: String, // explicitly required named prop
+      #[props(extends = GlobalAttributes)]
+      attrs: Vec<Attribute>,
+  ) -> Element {
+      rsx! {
+          div {
+              ..attrs, // forwarded HTML attributes
+              h2 { "{title}" }, // uses the explicit prop
+              p { "This is a card." }
+          }
+      }
+  }
+  ```
+
+- **Avoid extending incompatible element types**: You must **only extend attribute sets that match the tag you're rendering**. Extending `input` attributes on a `div` may compile, but attributes like `type` or `value` will be meaningless or even silently dropped in the browser.
+
+- **Prop filtering is manual**: If you want to restrict which attributes are allowed or filter out certain ones (e.g., to enforce accessibility or prevent accidental overrides), you must do this manually inside your component logic.
+
+- **Attribute order matters in practice**: While RSX generally follows a "last wins" policy, if you're conditionally overriding attributes (like `class`), you may want to control placement in the tree:
+
+  ```rust, no_run
+  rsx! {
+      div {
+          class: "default",
+          ..attrs, // if attrs contains a 'class', it will override "default"
+      }
+  }
+  ```
+
+- **Spreading is shallow**: `..attrs` simply forwards attribute key/value pairs. If you need to merge structured data (e.g., two `style` maps), you must handle it manually.
+
+- **Debugging tip**: When using `..attrs`, it can be hard to trace where an unexpected attribute came from. Consider logging or printing props during development to confirm which attributes were forwarded.
```

Comment:
I don't think this adds anything. The logging guide explains logging and we can assume people have basic debugging skills

Reply:
The statement is clear, but we don't need to document the fact that you can log things for every API. There isn't anything special about logging spread attributes, signals, events, etc. It is true, but it just bloats the documentation, which makes it more difficult to skim through, search, and maintain

Reply:
Thanks for the comments and the overall review. I learned some new things about `signal`s/`memo`s that I wasn't aware of. I'll take the above points into consideration and work on improving the documentation as a whole.